my_enums.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+
+module Enumerable #:nodoc: all
+  def my_each
+    length.times do |n|
+      current = self[n]
+      yield(current)
+    end
+  end
+
+  def my_each_with_index
+    length.times do |n|
+      index = n
+      current = self[n]
+      yield(index, current)
+    end
+  end
+
+  def my_select
+    emp_arr = []
+    my_each do |n|
+      emp_arr << n if yield(n)
+    end
+    emp_arr
+  end
+
+  def my_all?
+    emp_arr = []
+    my_each do |n|
+      emp_arr << n if yield(n)
+    end
+    emp_arr == self
+  end
+
+  def my_any?
+    tester = false
+    my_each do |n|
+      tester = true if yield(n)
+      break if tester
+    end
+    tester
+  end
+
+  def my_none?
+    tester = true
+    my_each do |n|
+      tester = false if yield(n)
+      break if tester == false
+    end
+    tester
+  end
+
+  def my_count(item = 'CANTOR')
+    count = 0
+    if item == 'CANTOR'
+      count = length
+    else
+      my_each do |n|
+        count += 1 if n == item
+      end
+    end
+    count
+  end
+
+  def my_map_one
+    emp_arr = []
+    my_each do |n|
+      emp_arr << yield(n)
+    end
+    emp_arr
+  end
```

Comment:
- [ ] enumerables my_map returns a new array with the results of running block once for every element in enum.
- [ ]  enumerables my_map returns an Enumerator if no block is given